pkg/http/utils.go

```diff
@@ -77,17 +85,93 @@ func ToHttpRequest(requestFlag string) (Request, error) {
 	}, nil
 }
 
-func interpolateDates(source string) string {
-	return todayTemplateRegex.ReplaceAllStringFunc(source, func(templateString string) string {
-		offsetDays := 0
+func legacyDateElements(source string) string {
+	offsetDays := 0
 
-		if templateString == "{tomorrow}" {
-			offsetDays = 1
-		} else if extractedOffset := todayTemplatePlusMinusRegex.FindString(templateString); len(extractedOffset) > 0 {
-			offsetDays, _ = strconv.Atoi(extractedOffset)
-		}
+	if source == "{today}" {
+		offsetDays = 0
+	} else if source == "{tomorrow}" {
+		offsetDays = 1
+	} else if extractedOffset := templatePlusMinusRegex.FindString(source); len(extractedOffset) > 0 {
+		offsetDays, _ = strconv.Atoi(extractedOffset)
+	}
+
+	// the date below is how the golang date formatter works. it's used for the formatting. it's not what is actually going to be displayed
+	return time.Now().Add(time.Duration(offsetDays) * 24 * time.Hour).Format("2006-01-02")
+}
+
+func dateElements(source string) string {
+	r := templateDatesRegex.FindStringSubmatch(source)
+
+	if r == nil {
+		return source
+	}
+	days := r[1]
+	months := r[2]
+	years := r[3]
+
+	offsetDays, _ := strconv.Atoi(days)
+	offsetMonths, _ := strconv.Atoi(months)
+	offsetYears, _ := strconv.Atoi(years)
+
+	// the date below is how the golang date formatter works. it's used for the formatting. it's not what is actually going to be displayed
+	return time.Now().AddDate(offsetYears, offsetMonths, offsetDays).Format("2006-01-02")
+}
+
+func timestampElements() string {
+	epoch := time.Now().UnixNano() / 1000000
+
+	return strconv.FormatInt(epoch, 10)
+}
+
+func randomElements(source string) string {
+	r := templateElementsRegex.FindStringSubmatch(source)
 
-		// the date below is how the golang date formatter works. it's used for the formatting. it's not what is actually going to be displayed
-		return time.Now().Add(time.Duration(offsetDays) * 24 * time.Hour).Format("2006-01-02")
+	if r == nil {
+		return source
+	}
+
+	s := strings.Split(r[1], ",")
+	number := rand.Intn(len(s))
+
+	return s[number]
+}
+
+func rangeElements(source string) string {
+	r := templateRangeRegex.FindStringSubmatch(source)
+	if r == nil {
+		return source
+	}
+
+	min, _ := strconv.Atoi(r[1])
+	max, _ := strconv.Atoi(r[2])
+
+	if min > max {
+		log.Printf("Invalid range. min > max")
+		return source
+	}
+
+	number := rand.Intn(max-min+1) + min
+
+	return strconv.Itoa(number)
+}
+
+func interpolatePlaceholders(source string) string {
+	return templatePlaceholderRegex.ReplaceAllStringFunc(source, func(templateString string) string {
+		rand.Seed(time.Now().UnixNano())
+
+		if strings.Contains(templateString, "today") || strings.Contains(templateString, "tomorrow") {
```

Comment:
If I understand this correctly users won't be able to use multiple placeholders as part of the same body, like a random range + a timestamp?

Reply:
they will. i added a test to cover multiple ones

Reply:
this inner function runs for every match of the templatePlaceholderRegex which captures all `{{ something }}` elements

Reply:
Ah cool, thanks